rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/ValidateXmlFilesRuleProvider.java

```diff
@@ -85,139 +72,198 @@ public Configuration getConfiguration(GraphContext context)
+
+                    @Override
+                    public InputSource resolveEntity(String publicId, String systemId) throws IOException, SAXException
+                    {
+                        return entityResolver.resolveEntity(publicId, systemId);
+                    }
+
+                    @Override
+                    public InputSource resolveEntity(String name, String publicId, String baseURI, String systemId) throws SAXException, IOException
+                    {
+                        return entityResolver.resolveEntity(name, publicId, baseURI, systemId);
+                    }
+
+                    @Override
+                    public void setDocumentLocator(Locator locator)
+                    {
+                        this.locator = locator;
+                    }
+
+                    @Override
+                    public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException
+                    {
+                        if (!firstElementFound)
+                        {
+                            firstElementFound = true;
+                            try
+                            {
+                                xsdURLs.addAll(getXSDLocations(attributes));
+
+                                // validate the xsd urls
+                                for (String xsdUrl : xsdURLs)
+                                {
+                                    // this will throw if it invalid
+                                    try  (InputStream is = new URL(xsdUrl).openStream())
+                                    {
+                                    }
+                                    catch (IOException e)
+                                    {
+                                        throw new InvalidXSDURLException(e.getMessage(), xsdUrl);
+                                    }
+                                }
+                            }
+                            catch (InvalidXSDURLException e)
+                            {
+                                parseExceptions.add(new SAXParseException(e.getMessage(), locator, e));
+                            }
+                        }
+                        super.startElement(uri, localName, qName, attributes);
+                    }
+
+                    @Override
+                    public void warning(SAXParseException e) throws SAXException
+                    {
+                        super.warning(e);
+                    }
+
+                    @Override
+                    public void error(SAXParseException e) throws SAXException
+                    {
+                        parseExceptions.add(e);
+                    }
+
+                    @Override
+                    public void fatalError(SAXParseException e) throws SAXException
+                    {
+                        parseExceptions.add(e);
+                    }
+                };
+
```

Comment:
Yeah, this class is starting to get more complicated. Will revisit if we hit more bugs.